telematrix/__init__.py

```diff
@@ -292,14 +292,14 @@ def get_username(user_id):
                     db.session.add(sender)
 
                     msg = None
-                    if 'unsigned' in event and 'prev_content' in event['unsigned']:
-                        prev = event['unsigned']['prev_content']
-                        if prev['membership'] == 'join':
-                            if 'displayname' in prev and prev['displayname']:
-                                oldname = prev['displayname']
-
-                            msg = '> {} changed their display name to {}'\
-                                  .format(oldname, displayname)
+                    print("DEBUG: {}".format(event))
+                    if ('unsigned' in event and 'prev_content' in event['unsigned']) or 'prev_content' in event:
```

Comment:
You seem to be checking for the same thing multiple times, mind rewriting this if statement to be a bit more efficient?